### PR TITLE
feat(RHINENG-1436, RHINENG-1427): Add group name filter to /resource-types/inventory-groups endpoint

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-1436](https://issues.redhat.com/browse/RHINENG-1436).
Adds the `name` filter to the `GET /resource-types/inventory-groups` endpoint. This supports the RBAC UI so that RBAC admins can more easily find groups to grant access to.

I also added an RBAC test to make sure RBAC admins could access the appropriate endpoints. That's more related to [RHINENG-1427](https://issues.redhat.com/browse/RHINENG-1427), but we should have the test, and I haven't yet been able to reproduce the issue described in that Jira.

Edit: I also included a fix to address [RHINENG-1427](https://issues.redhat.com/browse/RHINENG-1427), which is passing the app name through to the rbac request. It's only used when the user needs RBAC admin permission

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by fellipeh — APPROVED** [review]
lgtm

File: tests/test_api_resource_types_get.py [hunk 1]
```diff
@@ -61,3 +62,35 @@ def test_resource_types_groups_pagination(api_get, db_create_group, subtests):
...
+        assert group_result["id"] not in excluded_group_id_list
+
+    assert_resource_types_pagination(response_data, 1, 10, 1, "/inventory/v1/resource-types/inventory-groups")
+
+
+@pytest.mark.parametrize(
+    "url_builder",
+    [build_resource_types_url, build_resource_types_groups_url],
+)
+def test_get_resource_types_RBAC_allowed(mocker, api_get, url_builder, enable_rbac):
```
Comment (thearifismail) [inline tests/test_api_resource_types_get.py:87]: Are there some not-allowed permissions? i.e. should there be a test attempting forbidden actions?

**Review by thearifismail — COMMENTED** [review]
Minor comments.
General comment:  constants are strewn all over.  How about putting all constants and URLs in one file.  If all are in one place, then they will be named in a way that will communicate their intent leading to more clear code.

File: lib/middleware.py [hunk 2]
```diff
@@ -26,35 +26,35 @@
...
+ROUTE = "/api/rbac/v1/access/?application="
 CHECKED_TYPE = IdentityType.USER
 RETRY_STATUSES = [500, 502, 503, 504]
 
 outbound_http_metric = outbound_http_response_time.labels("rbac")
 
 
-def rbac_url():
-    return inventory_config().rbac_endpoint + ROUTE
+def rbac_url(app: str) -> str:
```
[the remark this answers is not in the document]
Comment (kruai) [reply lib/middleware.py:36]: OK, I can rename this while I'm at it

File: lib/middleware.py [hunk 3]
```diff
@@ -26,35 +26,35 @@
 
 logger = get_logger(__name__)
 
-ROUTE = "/api/rbac/v1/access/?application=inventory"
+ROUTE = "/api/rbac/v1/access/?application="
```
[the remark this answers is not in the document]
Comment (kruai) [reply lib/middleware.py:29]: I can rename this if you want, but APP_ROUTE is incorrect. If you prefer RBAC_ROUTE I can do that

File: tests/test_api_resource_types_get.py (hunk 1 above)
Comment (kruai) [reply tests/test_api_resource_types_get.py:87]: Yes, I can add a test for not-allowed permissions.

**Comment by kruai** [comment]
> Minor comments. General comment: constants are strewn all over. How about putting all constants and URLs in one file. If all are in one place, then they will be named in a way that will communicate their intent leading to more clear code.

@thearifismail Are you referring to RBAC constants, or constants in general? I think it's good to keep all middleware/rbac-related constants in the middleware file; it would be waaaaaaaaay messier to put every single constant we have in one file imo

File: lib/middleware.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (thearifismail) [reply lib/middleware.py]: this is definitely better than `APP_ROUTE`

[the remark this answers is not in the document]
Comment (kruai) [reply lib/middleware.py]: Cool (I had already made the change)

**Review by thearifismail — APPROVED** [review]
LGTM

**Pull request merged**